index.js

```diff
@@ -1,29 +1,48 @@
 const passport     = require('passport');
 const ldapStrategy = require('passport-ldapauth');
+const allowedAttributes = [
+  'dn', 'controls', 'objectClass', 'ou', 'o', 'ibmSerialNumber', 'employeeCountryCode', 
+  'isManager', 'managerCountryCode', 'co', 'uid', 'c', 'serialNumber', 'hrActive', 
+  'hrCompanyCode', 'hrCountryCode', 'hrFirstName', 'hrManagerIndicator', 'emailAddress', 
+  'mail', 'notesEmail', 'notesId', 'notesMailDomain', 'notesShortName', 'primaryNode', 'primaryUserId', 
+  'infotelephonenumber', 'personaltitle', 'jobresponsibilities', 'hrEmployeeType', 'employeeType', 
+  'preferredIdentity', 'iptelephonenumber', 'telephoneNumber', 'tieline', 'alternateTelephoneNumber', 
+  'mobile', 'timestampBpgui', 'hrLastName', 'coreDataIntegrity', 'directoryAlias', 'odmEntry', 'div', 
+  'notesMailFile', 'notesMailServer', 'buildingName', 'floor', 'ibmLoc', 'physicalDeliveryOfficeName', 
+  'workLoc', 'workLocation', 'workplaceIndicator', 'givenName', 'sn', 'timestampFeed', 'costCenter', 
+  'dept', 'divDept', 'hrOrganizationCode', 'callupName', 'cn', 'entryType', 'hrOrgUnitCode', 'manager', 
+  'managerSerialNumber', 'timestampHRMS', 'preferredLastName', 'timestampBpguiRI', 'passwordIsExpired', 
+  'passwordisreset', 'passwordIsResetByAdmin', 'passwordIsStruckOut', 'passwordModifyTimestamp']
 
 // Setup passport with IBM's LDAP config
 const config = { server: {
     url: 'ldaps://bluepages.ibm.com:636',
     searchBase: 'ou=bluepages,o=ibm.com',
-    searchFilter: '(mail={{username}})',
-    searchAttributes: ['cn', 'uid', 'mail']
+    searchFilter: '(mail={{username}})'
 }}
-passport.use(new ldapStrategy(config))
-passport.initialize()
 
-module.exports = function(req, res, next) {
+module.exports = {
+  initialize: function(attributes) {
+    if (attributes) {
+      let tmpAttributes = []
+      for (let attribute of attributes) {
+        if (allowedAttributes.indexOf(attribute) > 0) { tmpAttributes.push(attribute) }
+      }
+      config.server.searchAttributes = tmpAttributes
+    }
+
+    passport.use(new ldapStrategy(config))
+    passport.initialize()
+  },
+
+  authenticate: function(req, res, next) {
     passport.authenticate('ldapauth', {session: false}, (err, user, info) => {
-		if (err || !user)
-        	return res.sendStatus(401)
-      	
-      	// Sometimes user and mail can be arrays. Choose the first option
-      	// instead and make it a string
-        if (Array.isArray(user.cn) && user.cn.length > 0)
-			user.cn = user.cn[0]
-        if (Array.isArray(user.mail) && user.mail.length > 0)
-			user.mail = user.mail[0]
+      if (err || !user) { return res.sendStatus(401) }
+      if (Array.isArray(user.cn) && user.cn.length > 0) { user.cn = user.cn[0] }
```

Comment:
Well, what I want to do is guarantee that the module's response is always dependable on, meaning they will always get the same type when using `user.mail`, be it a string or 1 position array.

But I guess that might not be the point of this specific pull request (sorry about that). Since that's not a breaking change (`mail` will still be a string, there will only be a new field), we can do that with a minor update later on.